hack/make.ps1

```diff
@@ -0,0 +1,59 @@
+$DOCKER_PKG='github.com/docker/docker'
+$ErrorActionPreference = "Stop"
+
+# convert the .go-autogen bash script to ps1
+function ConvertGoAutogen {
+Write-Host "Converting hack\make\.go-autogen to hack\make\.go-autogen.ps1"
+(Get-Content hack\make\.go-autogen) `
+  -replace("rm -rf autogen", "if (Test-Path autogen) { rm -Recurse -Force autogen }
+  `$GITCOMMIT = ""$(git rev-parse --short HEAD)""
+  `$VERSION = ""$(cat VERSION)""
+  `$BUILDTIME = ""$(date)""
+  `$IAMSTATIC = ""false""
+  `$DOCKER_INITSHA1 = """"
+  `$DOCKER_INITPATH = """"
+  `$BASH_SOURCE = ""hack\make\.go-autogen.ps1""
+  ") `
+  -replace("cat > dockerversion/version_autogen.go <<DVEOF", '$version_autogen = @"') `
+  -replace(":-true", '') `
+  -replace("DVEOF", '"@ | Out-File -Encoding "Ascii" dockerversion\version_autogen.go') `
+  -replace ("cat > autogen/winresources/resources.go <<WREOF", '$resources = @"') `
+  -replace("WREOF", '"@ | Out-File -Encoding "Ascii" autogen/winresources\resources.go') `
+  -replace("if \[", '# if [') `
+  -replace("\\$", "``") `
+  -replace("> /dev/null", '') `
+  -replace('^fi', '# fi') `
+  | Out-File -Encoding "Ascii" hack\make\.go-autogen.ps1
+}
+
+# We're a nice, sexy, little shell script, and people might try to run us;
+# but really, they shouldn't. We want to be in a container!
+if ( ($pwd).ToString().Replace("\", "/") -ne "C:/go/src/$DOCKER_PKG") {
+  Write-Host "# WARNING! I don't seem to be running in the Docker container."
+  if ( [environment]::OSVersion.Version.Build -lt 10586 ) {
+    Write-Error "# ERROR! You can run this only on Windows Server 2016 TP4 or newer!"
+    exit 1
+  }
+  Write-Host "# Building Windows container"
+  docker build -t docker -f Dockerfile.windows .
+  if (Test-Path bundles) {
+    Write-Host "# bundles already exists. Removing"
+    Remove-Item bundles -Force -Recurse
+  }
+  mkdir bundles
+  Write-Host "# Running the build inside the Windows container"
+  docker run -it -v ${pwd}\bundles:c:/go/src/github.com/docker/docker/bundles docker "powershell -file hack\make.ps1"
+} else {
+  Write-Host "# Running inside the container ..."
+
+  Write-Host "# Converting hack/make/.go-autogen ..."
+  ConvertGoAutogen
+  Write-Host "# Running go-autogen.ps1 ..."
+  hack\make\.go-autogen.ps1
+
+  Write-Host "# Building docker binary ..."
+  $BUILDTAGS="autogen daemon"
+  go build -o ./docker.exe -tags "$BUILDTAGS" -ldflags "-linkmode=internal" ./docker
```

Comment:
Hmmm- it should work to a volume. Also, I'd go with the current think of the output filename having the VERSION in it, and linking back to docker.exe